tensorbay/client/requests.py

```diff
@@ -317,17 +323,73 @@ def multithread_upload(
     Arguments:
         function: The upload function.
         arguments: The arguments of the upload function.
+        callback_function: The callback function.
         jobs: The number of the max workers in multi-thread uploading procession.
         pbar: The :class:`Tqdm` instance for showing the upload process bar.
 
     """
     with ThreadPoolExecutor(jobs) as executor:
+        if callback_function is not None:
+            multi_callback = MultiCallbackMission(
+                function=function, callback_function=callback_function
+            )
+            function = multi_callback.mission  # type: ignore[assignment]
         futures = [executor.submit(function, argument) for argument in arguments]
+
         for future in futures:
             future.add_done_callback(pbar.update_callback)
 
         done, not_done = wait(futures, return_when=FIRST_EXCEPTION)
+
+        if callback_function is not None:
+            callback_arguments = [
+                multi_callback.arguments.get() for _ in range(multi_callback.arguments.qsize())
+            ]
+            if len(callback_arguments) > 0:
+                multi_callback.callback_function(callback_arguments)
+
         for future in not_done:
             future.cancel()
         for future in done:
             future.result()
+
+
+class MultiCallbackMission:  # pylint: disable=too-few-public-methods
+    """A class for callbacking in multi-thread work.
+
+    Arguments:
+        function: The function of a single thread.
+        callback_function: The callback function.
+
+    """
+
+    def __init__(
+        self, *, function: Callable[[_T], _R], callback_function: Callable[[List[_R]], None]
+    ) -> None:
+        self.lock = threading.Lock()
+        self.function = function
+        self.callback_function = callback_function
+        self.arguments: Queue[Any] = Queue()  # pylint: disable=unsubscriptable-object
+
+    def count_and_callback(self, callback: _R) -> None:
```

Comment:
the argument is not a callback, is return value from callback.

Reply:
I rename it with "callback_info".